src/framework/bundle/bundle.js

```diff
@@ -1,23 +1,59 @@
+import { EventHandler } from '../../core/event-handler.js';
+
 /**
- * Represents the resource of a Bundle Asset, which contains an index that maps URLs to blob URLs.
+ * Represents the resource of a Bundle Asset, which contains an index that maps URLs to DataViews.
  *
  * @ignore
  */
-class Bundle {
+class Bundle extends EventHandler {
+    /**
+     * Index of file url to to DataView.
+     * @type {Map<string, DataView>}
+     * @private
+     */
+    _index = new Map();
+
+    /**
+     * If Bundle has all files loaded.
+     * @type {boolean}
+     * @private
+     */
+    _loaded = false;
+
+    /**
+     * Fired when a file has been added to a Bundle.
+     *
+     * @event Bundle#add
+     * @param {string} url - A url of a file
+     * @param {DataView} data - A DataView of a file
+     * @example
+     * bundle.on("add", function (url, data) {
+     *     console.log("file added: " + url);
+     * });
+     */
+
     /**
-     * Create a new Bundle instance.
+     * Fired when all files of a Bundle has been loaded.
      *
-     * @param {object[]} files - An array of objects that have a name field and contain a
-     * getBlobUrl() function.
+     * @event Bundle#load
+     * @example
+     * bundle.on("load", function () {
+     *     console.log("All Bundle files has been loaded");
+     * });
      */
```

Comment:
define each event name as a static property on the class like the rest of the engine using the form `EVENT_<name>`